buildSrc/src/main/groovy/org/opensearch/gradle/plugin/PluginBuildPlugin.groovy

```diff
@@ -97,6 +98,9 @@ class PluginBuildPlugin implements Plugin<Project> {
             if (extension1.classname == null) {
                 throw new InvalidUserDataException('classname is a required setting for opensearchplugin')
             }
+            if (extension1.folderName == null) {
```

Comment:
Why is folder defaulted to a empty string and not left alone as `null`?

Reply:
The folder name property is added in `plugin-descriptor.properties` so when copying these properties over, the logic needs some value of folder name, by default String gets converted to null and it is not able to read that. Other optional properties like `extendedPlugins` also have a default added but it is a list so an empty list works for that. However, I have moved this default for folder name to `PluginPropertiesExtension.java` so the check here would not be needed.

Reply:
You could fix this by assigning the value in the constructor. Then you don't even need a method that returns `id` or `folderName` depending on whether the latter is set.